pkg/vm/memory/memory_test.go

```diff
@@ -277,6 +278,73 @@ func TestMemorySegmentsLoadDataTwoElements(t *testing.T) {
 	}
 }
 
+func TestValidateExistingMemoryForRangeCheckWithinBounds(t *testing.T) {
+	builtin := builtinrunner.NewRangeCheckBuiltinRunner(true)
+	segments := memory.NewMemorySegmentManager()
+	builtin.InitializeSegments(&segments)
+	builtin.AddValidationRule(&segments.Memory)
+
+	for i := 0; i < 3; i++ {
+		segments.AddSegment()
+	}
+	addr := memory.NewRelocatable(0, 0)
+	val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(45))
+	segments.Memory.Insert(addr, val)
+	segments.Memory.ValidateAddress(addr)
+	if !segments.Memory.ValidatedAdresses.Contains(addr) {
```

Comment:
We don't need to check the ValidatedAddresses, if the address validation fails then insertion will fail and we can check that. (We dont need to make ValidatedAddresses a public field because of this)

Reply:
Copied from: [7fb9ab9e00d70900a6da440d9a02d86326b9a750](https://github.com/lambdaclass/cairo-vm/commit/7fb9ab9e00d70900a6da440d9a02d86326b9a750#diff-f9eb247281a8f35a2e92707ac46de9e2133f3da1414d40a53194b3d15049d665R221)

Reply:
My comment still holds